fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2023,6 +2044,98 @@ private Bundle reconstructResponseBundle(Bundle responseBundle,
         return responseBundle;
     }
 
+    /**
+     * Returns a list of Integers that provide the indices of the bundle entries associated with the specified http
+     * method.
+     *
+     * @param requestBundle
+     *            the request bundle
+     * @param httpMethod
+     *            the http method to look for
+     * @return
+     */
+    private List<Integer> getBundleRequestIndicesForMethod(Bundle requestBundle,
+        Bundle responseBundle, HTTPVerb httpMethod) {
+        List<Integer> indices = new ArrayList<>();
+        for (int i = 0; i < requestBundle.getEntry().size(); i++) {
+            Bundle.Entry requestEntry = requestBundle.getEntry().get(i);
+            Bundle.Entry.Request request = requestEntry.getRequest();
+
+            Bundle.Entry responseEntry = responseBundle.getEntry().get(i);
+            Bundle.Entry.Response response = responseEntry.getResponse();
+
+            // If the response status is SC_OK which means the request passed the validation,
+            // and this request entry's http method is the one we're looking for,
+            // then record the index in our list.
+            // (please notice that status can not be null since R4, So we set the response status as SC_OK
+            // after the resource validation. )
+            if (response.getStatus().equals(SC_OK_STRING)
+                    && request.getMethod().equals(httpMethod)) {
+                indices.add(Integer.valueOf(i));
+            }
+        }
+        return indices;
+    }
+
+    /**
+     * This function sorts the request entries in the specified bundle, based on the path part of the entry's 'url'
+     * field.
+     *
+     * @param bundle
+     *            the bundle containing the request entries to be sorted.
+     * @return an array of Integer which provides the "sorted" ordering of request entry index values.
+     */
+    private void sortBundleRequestEntries(Bundle bundle, List<Integer> indices) {
+        // Sort the list of indices based on the contents of their entries in the bundle.
+        Collections.sort(indices, new BundleEntryComparator(bundle.getEntry()));
+    }
+
+    private static class BundleEntryComparator implements Comparator<Integer> {
+        private List<Bundle.Entry> entries;
+
+        public BundleEntryComparator(List<Bundle.Entry> entries) {
+            this.entries = entries;
+        }
+
+        @Override
+        public int compare(Integer indexA, Integer indexB) {
+            Bundle.Entry a = entries.get(indexA);
+            Bundle.Entry b = entries.get(indexB);
+            String pathA = getUrlPath(a);
+            String pathB = getUrlPath(b);
+
+            log.fine("Comparing request entry URL paths: " + pathA + ", " + pathB);
```

Comment:
Need if isLoggable barrier here

Reply:
like this:
```suggestion
            if (log.isLoggable(Level.FINE)) {
                log.fine("Comparing request entry URL paths: " + pathA + ", " + pathB);
            }
```

Reply:
Added check around all log.fine() statements that did not have check.

Reply:
FWIW, I don't think its always needed.  But it never hurts to have it, so I'm OK with adding it everywhere.